components/atom/input/src/Input/Features/WithIcon/index.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable react/prop-types, no-unused-vars, no-console */
+
+import React from 'react'
+import PropTypes from 'prop-types'
+
+const CLASS_ICON = 'sui-AtomInput-withIcon'
+const CLASS_ICON_COMPONENT = `${CLASS_ICON}--icon`
+
+const WithIconHoC = WrappedInput =>
+  class Addon extends React.Component {
+    static propTypes = {
```

Comment:
I like to define props this way better. But I thought we had a differente convention that implies to put it in bottom.